kv/range_cache.go

```diff
@@ -149,7 +162,10 @@ func (rmc *RangeDescriptorCache) EvictCachedRangeDescriptor(key proto.Key) {
 // the cache.
 func (rmc *RangeDescriptorCache) getCachedRangeDescriptor(key proto.Key) (
 	rangeCacheKey, *proto.RangeDescriptor) {
-	metaKey := engine.RangeMetaKey(key)
+	// We want to look up the range descriptor for key. The cache is
+	// indexed using the end-key of the range, but the end-key is
+	// non-inclusive. So we access the cache using key.Next().
```

Comment:
Why do we use EndKey and Ceil instead of StartKey and Floor? Indexing by start key would avoid this issue. (I'm sure there's a reason but it doesn't appear to be explained anywhere)